components/parameters-sidebar.tsx

```diff
@@ -26,8 +28,20 @@ interface ParametersSidebarProps {
   onChange: (key: keyof Parameters, value: number | boolean | string) => void;
 }
 
-// Replace the entire content of ParametersSidebar
 export function ParametersSidebar({ parameters, onChange }: ParametersSidebarProps) {
+  const handleSave = () => {
+    try {
+      // Creates a copy of parameters without the API key
+      const parametersToSave = { ...parameters };
+      delete parametersToSave.apiKey;
+
+      localStorage.setItem("playground-parameters", JSON.stringify(parametersToSave));
```

Comment:
Whenever you call updatePlaygroundSettings it is saved to localstorage. you don't have to do it here.